File: fix_ports.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    file = path.join(dir, file);
    const stat = fs.statSync(file);
    if (stat && stat.isDirectory()) results = results.concat(walk(file));
    else if (file.endsWith('.js') || file.endsWith('.html')) results.push(file);
  });
  return results;
}

const dirs = ['./Frontend/js', './Frontend/pages'];
dirs.forEach(dir => {
  if (!fs.existsSync(dir)) return;
  const files = walk(dir);
  files.forEach(file => {
    const content = fs.readFileSync(file, 'utf8');
    const newContent = content.replace(/(localhost|127\.0\.0\.1):5000/g, 'localhost:5001');
    if (content !== newContent) {
      fs.writeFileSync(file, newContent, 'utf8');
      console.log(`Updated ${file}`);
    }
  });
});

File: fix_ports.py
import os
import re

def fix_port(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.js', '.html')):
                path = os.path.join(root, file)
                try:
                    with open(path, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    new_content = re.sub(r'(localhost|127\.0\.0\.1):5000', r'localhost:5001', content)
                    
                    if content != new_content:
                        with open(path, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        print(f"Updated {path}")
                except Exception as e:
                    print(f"Error processing {path}: {e}")

if __name__ == "__main__":
    fix_port('Frontend/js')
    fix_port('Frontend/pages')
